source/particle/property/cpo_bingham_average.cc

```diff
@@ -0,0 +1,329 @@
+
+
+        const Tensor<1,3,double> averaged_a = eigenvectors_a[0].second * eigenvectors_a[0].first;
+        const Tensor<1,3,double> averaged_b = eigenvectors_b[0].second * eigenvectors_b[0].first;
+        const Tensor<1,3,double> averaged_c = eigenvectors_c[0].second * eigenvectors_a[0].first;
+
+        return
+        {
+          {
+            {{averaged_a[0],averaged_a[1],averaged_a[2]}},
+            {{averaged_b[0],averaged_b[1],averaged_b[2]}},
+            {{averaged_c[0],averaged_c[1],averaged_c[2]}}
+          }
+        };
+      }
+
+      template<int dim>
+      template <typename T>
+      std::vector<std::size_t>
+      CpoBinghamAverage<dim>::sort_permutation(
+        const std::vector<T> &vec) const
+      {
+        std::vector<std::size_t> p(vec.size());
+        std::iota(p.begin(), p.end(), 0);
+        std::sort(p.begin(), p.end(),
+                  [&](std::size_t i, std::size_t j)
+        {
+          return vec[i] < vec[j];
+        });
+        return p;
+      }
+
+      template<int dim>
+      template <typename T>
+      std::vector<T>
+      CpoBinghamAverage<dim>::apply_permutation(
+        const std::vector<T> &vec,
+        const std::vector<std::size_t> &p) const
+      {
+        std::vector<T> sorted_vec(vec.size());
+        std::transform(p.begin(), p.end(), sorted_vec.begin(),
+                       [&](std::size_t i)
+        {
+          return vec[i];
+        });
+        return sorted_vec;
+      }
+
+
+
+      template <int dim>
+      UpdateTimeFlags
+      CpoBinghamAverage<dim>::need_update() const
+      {
+        return update_output_step;
+      }
+
+      template <int dim>
+      UpdateFlags
+      CpoBinghamAverage<dim>::get_needed_update_flags () const
+      {
+        return update_default;
+      }
+
+      template <int dim>
+      std::vector<std::pair<std::string, unsigned int>>
+      CpoBinghamAverage<dim>::get_property_information() const
+      {
+        std::vector<std::pair<std::string,unsigned int>> property_information;
+        for (unsigned int mineral_i = 0; mineral_i < n_minerals; mineral_i++)
+          {
+            property_information.push_back(std::make_pair("cpo mineral " + std::to_string(mineral_i) + " bingham average a axis",3));
+            property_information.push_back(std::make_pair("cpo mineral " + std::to_string(mineral_i) + " bingham average b axis",3));
+            property_information.push_back(std::make_pair("cpo mineral " + std::to_string(mineral_i) + " bingham average c axis",3));
+          }
+
+        return property_information;
+      }
+
+      template <int dim>
+      void
+      CpoBinghamAverage<dim>::declare_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Particles");
+          {
+            prm.enter_subsection("CpoBinghamAverage");
```

Comment:
To be consistent with our usual naming scheme of subsection this would need to  be:
```suggestion
            prm.enter_subsection("CPO Bingham Average");
```